doc/source/serve/dev-workflow.md

```diff
@@ -0,0 +1,130 @@
+
+Recall our example above:
+
+```{literalinclude} ../serve/doc_code/local_dev.py
+:start-after: __local_dev_start__
+:end-before: __local_dev_end__
+:language: python
+```
+
+Now run the following command in your terminal:
+
+```bash
+serve run local_dev:HelloDeployment
+```
+
+This will output something similar to the following:
+
+```console
+2022-08-11 11:31:47,692 INFO scripts.py:294 -- Deploying from import path: "local_dev:graph".
+2022-08-11 11:31:50,372 INFO worker.py:1481 -- Started a local Ray instance. View the dashboard at http://127.0.0.1:8265.
+(ServeController pid=9865) INFO 2022-08-11 11:31:54,039 controller 9865 http_state.py:129 - Starting HTTP proxy with name 'SERVE_CONTROLLER_ACTOR:SERVE_PROXY_ACTOR-dff7dc5b97b4a11facaed746f02448224aa0c1fb651988ba7197e949' on node 'dff7dc5b97b4a11facaed746f02448224aa0c1fb651988ba7197e949' listening on '127.0.0.1:8000'
+(ServeController pid=9865) INFO 2022-08-11 11:31:55,373 controller 9865 deployment_state.py:1232 - Adding 1 replicas to deployment 'Doubler'.
+(ServeController pid=9865) INFO 2022-08-11 11:31:55,389 controller 9865 deployment_state.py:1232 - Adding 1 replicas to deployment 'HelloDeployment'.
+(HTTPProxyActor pid=9872) INFO:     Started server process [9872]
+2022-08-11 11:31:57,383 SUCC scripts.py:315 -- Deployed successfully.
+```
+
+The `serve run` command blocks the terminal and can be canceled with Ctrl-C.
+
+Now that Serve is running, we can pass in HTTP requests to the application and see the output.
+For simplicity, we'll just use the `curl` command to send requests from another terminal.
+
+```bash
+curl -X PUT "http://localhost:8000/?name=Ray"   
+```
+
+This will output:
+
+```console
+Hello, Ray! Hello, Ray!
+```
+
+After you're done testing, you can shut down Ray Serve with Ctrl-C:
+
+```console
+^C2022-08-11 11:47:19,829       INFO scripts.py:323 -- Got KeyboardInterrupt, shutting down...
+(ServeController pid=9865) INFO 2022-08-11 11:47:19,926 controller 9865 deployment_state.py:1257 - Removing 1 replicas from deployment 'Doubler'.
+(ServeController pid=9865) INFO 2022-08-11 11:47:19,929 controller 9865 deployment_state.py:1257 - Removing 1 replicas from deployment 'HelloDeployment'.
+```
+
+## Testing on a remote cluster
+
+When making the transition from your local machine to a remote cluster, you'll need to make sure your cluster has a similar environment to your local machine--files, environment variables, and Python packages, for example.  During development, you can use {ref}`Runtime Environments<runtime-environments>` to manage this in a flexible way.
+
+See [Ray Client](ray-client-under-construction) for more information on the Ray address specified here by the `--address` parameter.
+
+Let's see a simple example.
+
+```bash
+serve run  --address=ray://<cluster-ip-address>:10001 --runtime-env-json='{"env_vars": {"MY_ENV_VAR": "my-value"}, "working_dir": "./project/src", "pip": ["requests", "chess"]}' local_dev:HelloDeployment
```

Comment:
Can we split this example into multiple steps to make it more digestible? Concrete suggestion:

1. First explain how to run Serve on a remote cluster, no runtime env involved.
2. Then show the common case that's currently mentioned at the end: "A common pattern is to use the root directory of your project as the `working_dir` of your runtime environment when testing on a remote cluster."
3. End with the advanced case where a full runtime env json / yaml is used.

Reply:
For step 1, I don't think there's a dev workflow that doesn't use runtime_env, because we need runtime_env to get the local code onto the cluster.  

Why don't I try this ordering to make it more digestible:
1. Just use `--working-dir`
2. Use a full `--runtime-env-json`
3. Mention runtime_env YAML

Reply:
Ah I see. Yes, that flow sounds good, thanks!

Reply:
I gave this a shot, let me know how it looks